### DISCUSSION TITLE
Using automatic code signing with CI (GitHub Actions), why can't Xcode find any Xcode accounts/provisioning profiles?

### DISCUSSION THREAD

**Discussion by wambambizzle** [fastlane/fastlane#18412] [Q&A] [answered]
- [x] Updated fastlane to the latest version
 - [x] I read the Contribution Guidelines
 - [x] I read docs.fastlane.tools
 - [x] I searched for existing GitHub issues

I have been trying to get code signing working on GitHub Actions ci build for a few days (automatic and manaul) with no luck.

### Issue

error: There are no accounts registered with Xcode. Add your developer account to Xcode (in target 'Testx' from project 'Testx')

error: No profiles for 'com.example.local' were found: Xcode couldn't find any iOS App Development provisioning profiles matching 'com.example.local'. (in target 'Testx' from project 'Testx')

#### Command executed
`bundle exec fastlane build_workflow`

### GitHub Actions yml
```
name: build

# Controls when the action will run. 
on: [push]

jobs:
  build:
    name: Build
    runs-on: macos-latest
    timeout-minutes: 30
    env:
      FASTLANE_USER: ${{ secrets.FASTLANE_USER }}
      FASTLANE_ITC_TEAM_ID: ${{secrets.FASTLANE_ITC_TEAM_ID}}
      FASTLANE_PASSWORD: ${{ secrets.FASTLANE_PASSWORD }}
      FASTLANE_APPLE_APPLICATION_SPECIFIC_PASSWORD: ${{ secrets.FASTLANE_APPLE_APPLICATION_SPECIFIC_PASSWORD }}
      FASTLANE_SESSION: ${{ secrets.FASTLANE_SESSION }}
      FASTLANE_APPLE_KEYCHAIN_PASSWORD: ${{secrets.APPLE_KEYCHAIN_PASSWORD}}
      KEYCHAIN_NAME: ${{secrets.KEYCHAIN_NAME}}

    steps:
      - name: Checkout
        uses: actions/checkout@v2

      - name: Select Xcode Version
        uses: maxim-lobanov/setup-xcode@v1
        with:
          xcode-version: latest-stable

      - name: Install and Update Bundle
        run: |
          gem install bundler:1.17.3
          bundle update
      
      - name: Apple Developer Login
        run: bundle exec fastlane fastlane-credentials add --username FASTLANE_USER --password FASTLANE_PASSWORD

      - name: Download Xcode Profiles
        run: bundle exec fastlane sigh download_all

      - name: Test new fastlane build lane
        run: bundle exec fastlane build_workflow --verbose
```

### Fastlane Output
+-----------------------+-------------------------+
|        Summary for code signing settings        |
+-----------------------+-------------------------+
| targets               | ["Testx"]               |
| path                  | Testx.xcodeproj         |
| bundle_identifier     | com.test.local |
| use_automatic_signing | true                    |
+-----------------------+-------------------------+

Modified Targets:

*Testx
Modified Build Configurations:
* Debug
* QA
* Local
* QA2
* Release
--- Step: build_app ---
-----------------------

+--------------------------------------+-----------------------------------------------------------+
|                                     Summary for gym 2.178.0                                      |
+--------------------------------------+-----------------------------------------------------------+

| workspace                            | Testx.xcworkspace                                         |
|
| scheme                               | Testx Local                                               |
|
| configuration                        | Local                                                     |
|
| xcargs                               | -allowProvisioningUpdates                                 |
|
| clean                                | true                                                      |
|
| silent                               | true                                                      |
|
| export_method                        | ad-hoc                                                    |
|
| skip_archive                         | true                                                      |
|
| output_directory                     | .                                                         |
|
| output_name                          | Pods_Testx                                                |
|
| skip_package_ipa                     | false                                                     |
|
| skip_package_pkg                     | false                                                     |
|
| build_path                           | /Users/runner/Library/Developer/Xcode/Archives/2021-03-16 |
|
| result_bundle                        | false                                                     |
|
| buildlog_path                        | ~/Library/Logs/gym                                        |
|
| destination                          | generic/platform=iOS                                      |
|
| skip_profile_detection               | false                                                     |
|
| skip_package_dependencies_resolution | false                                                     |
|
| disable_package_automatic_updates    | false                                                     |
|
| use_system_scm                       | false                                                     |
|
| xcode_path                           | /Applications/Xcode_12.4.app                              |
+--------------------------------------+-----------------------------------------------------------+

+------------------------------------------------------------------+----------------------+
|                                 Generated Build Command                                 |
+------------------------------------------------------------------+----------------------+
| Option                                                           | Value                |
+------------------------------------------------------------------+----------------------+
| set -o pipefail &&                                               |                      |
|
| xcodebuild                                                       |                      |
|
| -workspace Testx.xcworkspace                                     |                      |
|
| -scheme Testx\ Local                                             |                      |
|
| -configuration Local                                             |                      |
|
| -destination                                                     | generic/platform=iOS |
|
| -allowProvisioningUpdates                                        |                      |
|
| clean                                                            |                      |
|
| build                                                            |                      |
|
| | tee /Users/runner/Library/Logs/gym/Pods_Testx-Testx\ Local.log |                      |
|
| | xcpretty                                                       |                      |
+------------------------------------------------------------------+----------------------+

** BUILD FAILED **
ERROR [2021-03-16 21:30:16.32]: Exit status: 65
INFO [2021-03-16 21:30:16.33]: 
WARN [2021-03-16 21:30:16.33]: Maybe the error shown is caused by using the wrong version of Xcode
WARN [2021-03-16 21:30:16.33]: Found multiple versions of Xcode in '/Applications/'
WARN [2021-03-16 21:30:16.34]: Make sure you selected the right version for your project
WARN [2021-03-16 21:30:16.34]: This build process was executed using '/Applications/Xcode_12.4.app'
WARN [2021-03-16 21:30:16.34]: If you want to update your Xcode path, either
INFO [2021-03-16 21:30:16.34]: 
INFO [2021-03-16 21:30:16.34]: - Specify the Xcode version in your Fastfile
INFO [2021-03-16 21:30:16.34]: ▸ xcversion(version: "8.1") # Selects Xcode 8.1.0
INFO [2021-03-16 21:30:16.34]: 
INFO [2021-03-16 21:30:16.34]: - Specify an absolute path to your Xcode installation in your Fastfile
INFO [2021-03-16 21:30:16.34]: ▸ xcode_select "/Applications/Xcode8.app"
INFO [2021-03-16 21:30:16.34]: 
INFO [2021-03-16 21:30:16.34]: - Manually update the path using
INFO [2021-03-16 21:30:16.34]: ▸ sudo xcode-select -s /Applications/Xcode.app
INFO [2021-03-16 21:30:16.34]: 

|
+---------------+------------------------------+
|
|              Build environment               |
|
+---------------+------------------------------+
|
| xcode_path    | /Applications/Xcode_12.4.app |
|
| gym_version   | 2.178.0                      |
|
| export_method | ad-hoc                       |
|
| sdk           | iPhoneOS14.4.sdk             |
|
+---------------+------------------------------+
```


### Environment
<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                            |
| --------------------------- | ------------------------------------------------ |
| OS                          | 11.2.1                                           |
| Ruby                        | 2.4.1                                            |
| Bundler?                    | true                                             |
| Git                         | git version 2.25.0                               |
| Installation Source         | ~/.rvm/gems/ruby-2.4.1@exchange-ios/bin/fastlane |
| Host                        | macOS 11.2.1 (20D74)                             |
| Ruby Lib Dir                | ~/.rvm/rubies/ruby-2.4.1/lib                     |
| OpenSSL Version             | OpenSSL 1.1.1d  10 Sep 2019                      |
| Is contained                | false                                            |
| Is homebrew                 | false                                            |
| Is installed via Fabric.app | false                                            |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/      |
| Xcode Version               | 12.4                                             |


### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE |             |   |


### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
default_platform(:ios)

platform :ios do
  desc "Test building for Github Actions"
  lane :build_workflow do

    set_keychain

    cocoapods(clean_install: true)                  # update pods
    clean_cocoapods_cache      # clean cocoapods
    clear_derived_data         # remove any local build data

    # use this to grab right scheme and configuration
    app_identifier = CredentialsManager::AppfileConfig.try_fetch_value(:app_identifier)
    scheme = get_scheme(app_identifier)
    configuration = get_configuration(app_identifier)
    workspace = "Testx.xcworkspace"

    update_code_signing_settings(
      targets: ["Testx"],
      path: "Testx.xcodeproj",
      bundle_identifier: app_identifier,
      use_automatic_signing: true
    )

    build_app(
      workspace: workspace,
      scheme: scheme,
      configuration: configuration,
      xcargs: "-allowProvisioningUpdates",
      clean: true,
      silent: true,
      export_method: "ad-hoc",
      skip_archive: true
    )

    delete_keychain(name: ENV["KEYCHAIN_NAME"])
  end

  # Helper Methods

  def get_configuration(identifier)
    case identifier
    when "com.example.qa"
      "QA"
    when "com.example.qa2"
      "QA2"
    when "com.example.qa3"
      "Debug"
    when "com.example.local"
      "Local"
    end
  end

  def get_scheme(identifier)
    case identifier
    when "com.example.qa"
      "QA"
    when "com.iexample.qa2"
      "QA2"
    when "com.example.qa3"
      "QA3"
    when "com.example.local"
      "Example Local"
    end
  end

  def set_keychain
    # TODO: do not run locally
    keychain_name = ENV["KEYCHAIN_NAME"]
    keychain_password = ENV["FASTLANE_APPLE_KEYCHAIN_PASSWORD"]

    create_keychain(
      name: keychain_name,
      password: keychain_password,
      default_keychain: true,
      unlock: true,
      timeout: 3600,
      lock_when_sleeps: false
    )
  end
```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
app_identifier("com.test.fastlane") # The bundle identifier of your app
apple_id("#####@email.com") # Your Apple email address

itc_team_id("#######") # App Store Connect Team ID
team_id("#######") # Developer Portal Team ID

for_platform :ios do
  for_lane :release_qa do
    app_identifier "com.example.qa"
  end

  for_lane :release_qa2 do
    app_identifier "com.example.qa2"
  end

  for_lane :release_qa3 do
    app_identifier "com.example.qa3"
  end

  for_lane :build_workflow do
    app_identifier "com.example.local"
  end
end
```
</details>

```
### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.178.0 | ✅ Up-To-Date  |


### Loaded fastlane plugins:

| Plugin                  | Version | Update-Status |
| ----------------------- | ------- | ------------- |
| fastlane-plugin-bugsnag | 2.0.0   | ✅ Up-To-Date  |


<details><summary><b>Loaded gems</b></summary>

| Gem                           | Version      |
| ----------------------------- | ------------ |
| did_you_mean                  | 1.1.0        |
| executable-hooks              | 1.6.0        |
| bundler-unload                | 1.0.2        |
| rubygems-bundler              | 1.4.5        |
| bundler                       | 1.17.3       |
| rake                          | 13.0.3       |
| CFPropertyList                | 3.0.3        |
| concurrent-ruby               | 1.1.6        |
| i18n                          | 0.9.5        |
| minitest                      | 5.14.1       |
| thread_safe                   | 0.3.6        |
| tzinfo                        | 1.2.7        |
| activesupport                 | 5.2.4.4      |
| public_suffix                 | 4.0.6        |
| addressable                   | 2.7.0        |
| httpclient                    | 2.8.3        |
| json                          | 2.5.1        |
| algoliasearch                 | 1.27.3       |
| artifactory                   | 3.0.15       |
| atomos                        | 0.1.3        |
| aws-eventstream               | 1.1.1        |
| aws-partitions                | 1.432.0      |
| aws-sigv4                     | 1.2.3        |
| jmespath                      | 1.4.0        |
| aws-sdk-core                  | 3.113.0      |
| aws-sdk-kms                   | 1.43.0       |
| aws-sdk-s3                    | 1.91.0       |
| babosa                        | 1.0.4        |
| claide                        | 1.0.3        |
| fuzzy_match                   | 2.0.4        |
| nap                           | 1.1.0        |
| netrc                         | 0.11.0       |
| ffi                           | 1.13.1       |
| ethon                         | 0.12.0       |
| typhoeus                      | 1.4.0        |
| cocoapods-core                | 1.10.1       |
| cocoapods-deintegrate         | 1.0.4        |
| cocoapods-downloader          | 1.4.0        |
| cocoapods-plugins             | 1.0.0        |
| cocoapods-search              | 1.0.0        |
| cocoapods-trunk               | 1.5.0        |
| cocoapods-try                 | 1.2.0        |
| colored2                      | 3.1.2        |
| escape                        | 0.0.4        |
| fourflusher                   | 2.3.1        |
| gh_inspector                  | 1.1.3        |
| molinillo                     | 0.6.6        |
| ruby-macho                    | 1.4.0        |
| nanaimo                       | 0.3.0        |
| xcodeproj                     | 1.19.0       |
| cocoapods                     | 1.10.1       |
| colored                       | 1.2          |
| highline                      | 1.7.10       |
| commander-fastlane            | 4.4.6        |
| declarative                   | 0.0.20       |
| declarative-option            | 0.1.0        |
| digest-crc                    | 0.6.3        |
| unf_ext                       | 0.0.7.7      |
| unf                           | 0.1.4        |
| domain_name                   | 0.5.20190701 |
| dotenv                        | 2.7.6        |
| emoji_regex                   | 3.2.2        |
| excon                         | 0.79.0       |
| faraday-net_http              | 1.0.1        |
| multipart-post                | 2.0.0        |
| ruby2_keywords                | 0.0.4        |
| faraday                       | 1.3.0        |
| http-cookie                   | 1.0.3        |
| faraday-cookie_jar            | 0.0.7        |
| faraday_middleware            | 1.0.0        |
| fastimage                     | 2.2.3        |
| jwt                           | 2.2.2        |
| memoist                       | 0.16.2       |
| multi_json                    | 1.15.0       |
| os                            | 1.1.1        |
| signet                        | 0.14.1       |
| googleauth                    | 0.15.1       |
| mini_mime                     | 1.0.2        |
| uber                          | 0.1.0        |
| representable                 | 3.0.4        |
| retriable                     | 3.1.2        |
| google-api-client             | 0.38.0       |
| rexml                         | 3.2.4        |
| webrick                       | 1.7.0        |
| google-apis-core              | 0.2.1        |
| google-apis-iamcredentials_v1 | 0.1.0        |
| google-apis-storage_v1        | 0.2.0        |
| google-cloud-env              | 1.4.0        |
| google-cloud-errors           | 1.0.1        |
| google-cloud-core             | 1.5.0        |
| google-cloud-storage          | 1.30.0       |
| mini_magick                   | 4.11.0       |
| naturally                     | 2.2.1        |
| plist                         | 3.6.0        |
| rubyzip                       | 2.3.0        |
| security                      | 0.1.3        |
| simctl                        | 1.6.8        |
| slack-notifier                | 2.3.2        |
| terminal-notifier             | 2.0.0        |
| unicode-display_width         | 1.7.0        |
| terminal-table                | 1.8.0        |
| tty-screen                    | 0.8.1        |
| tty-cursor                    | 0.7.1        |
| tty-spinner                   | 0.9.3        |
| word_wrap                     | 1.0.0        |
| rouge                         | 2.0.7        |
| xcpretty                      | 0.3.0        |
| xcpretty-travis-formatter     | 1.0.1        |
| rchardet                      | 1.8.0        |
| git                           | 1.8.1        |
| xml-simple                    | 1.1.8        |
| fastlane-plugin-bugsnag       | 2.0.0        |
</details>


*generated on:* **2021-03-17**
</details>
```

**Comment by GZaccaroni** [comment]
Same problem

**Comment by mayerlench** [comment]
Did you figure this issue?

**Comment by thitoo-yf** [comment]
Any idea for this issue. I have the same problem as your issue.

**Comment by wambambizzle** [comment, marked as answer]
Sorry for the wait! I did not see that I had any responses. IIRC my fast file only has two real changes. One to check if `is_ci` and run my key chain logic and the other is to `skip_codesigning: true` during `build_app` command.

I also removed the `Download Xcode Profiles` from my build yml from feedback I received that the CI builds are fresh so they won't have any idea about your projects provisioning profiles even if you down load them. Those steps did the trick for me.

If you're using cocoa pods at all you may need a step like this as well.

```
# This will use the version from .ruby-version
      - name: Set Ruby Version
        uses: ruby/setup-ruby@v1
        with:
          bundler-cache: true # runs 'bundle install' and caches installed gems automatically
```

Hope this helps, again sorry for the long delay!

As for skipping code signing during CI builds, I haven't seen any negative side effects at all. If my CI build breaks it has only been because of my code changes (nothing provisioning profile related).
